core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMEdge.kt

```diff
@@ -95,11 +99,27 @@ data class STDCMEdge(
                 newWaypointIndex,
                 envelopeStartOffset + length.distance,
                 stopDuration,
+                firstStopAfterIndex.plannedTimingData,
+                prevPlannedNodeTimeDiff,
                 graph.remainingTimeEstimator.invoke(this, locationOnEdge, newWaypointIndex)
             )
         }
     }
 
+    private fun getPreviousPlannedNodeTimeDiff(): Double? {
+        var prevPlannedNode = this.previousNode
+        while (prevPlannedNode != null) {
+            if (prevPlannedNode.plannedTimingData != null) {
+                return prevPlannedNode.plannedTimingData!!.arrivalTime.seconds -
+                    prevPlannedNode.time
+            } else if (prevPlannedNode.prevPlannedNodeTimeDiff != null) {
+                return prevPlannedNode.prevPlannedNodeTimeDiff
+            }
+            prevPlannedNode = prevPlannedNode.previousEdge.previousNode
+        }
+        return null
+    }
```

Comment:
I don't think this is right. We need to account for the departure time shift to know the arrival time at that location

Reply:
Yeah, we have to take into account the additional departure time shift added since that node. The previousPlanned node's actual arrival time should be equal to :
`prevPlannedNode.time - (currentNode.totalPrevAddedDelay - prevPlannedNode.totalPrevAddedDelay)`